pkg/util/kubernetes/hpa/hpa.go

```diff
@@ -11,13 +11,24 @@ import (
 	"reflect"
 
 	autoscalingv2 "k8s.io/api/autoscaling/v2beta1"
+	"k8s.io/api/core/v1"
 
 	"github.com/DataDog/datadog-agent/pkg/clusteragent/custommetrics"
 	"github.com/DataDog/datadog-agent/pkg/util/log"
 )
 
-// Inspect returns the list of external metrics from the hpa to use for autoscaling.
-func Inspect(hpa *autoscalingv2.HorizontalPodAutoscaler) (emList []custommetrics.ExternalMetricValue) {
+// IsAbleToScale returns if the HPA is currently able to scale (not in BackoffDownscale)
+func IsAbleToScale(hpa *autoscalingv2.HorizontalPodAutoscaler) bool {
+	for _, condition := range hpa.Status.Conditions {
+		if autoscalingv2.AbleToScale == condition.Type {
+			return condition.Status == v1.ConditionTrue
```

Comment:
I was trying to review [the possible statuses](https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/podautoscaler/horizontal.go#L224-L299), but I am afraid there might be an edge case here. If the metric is unavailable, the Cluster agent will not serve it as a result the HPA will move to a invalid state for the metric which would lead the cluster agent to never check the metric.
I did some testing on my end, I think adding a reason on top of the AbleToScale to true should suffice.
The reason in question we are interested in is `BackoffBoth`/`BackoffDownscale`/`BackoffUpscale ` but it's only in 1.10, 1.11! https://github.com/kubernetes/kubernetes/blob/v1.10.0/pkg/controller/podautoscaler/horizontal.go#L484

This was removed after: https://github.com/kubernetes/kubernetes/pull/66615

So it might cause issues with more recent version of K8s :(

Reply:
I realized that there is still logic for the downscale cooldown delayhttps://github.com/kubernetes/kubernetes/commit/958cba1c82e39dbbd25600128b9041660f80afc8#diff-ff4e8f1a0c24ee394a0ddf37d64b2960R572.
Let's keep that for the discussion.